### ISSUE TITLE
[TensorFlow Java] Generated factory methods for building operations do not take input list

### ISSUE THREAD

**Issue by irenedea** [tensorflow/tensorflow#24483]
**System information**
- TensorFlow version: 1.12

**Describe the current behavior**
There are automatically generated wrapper classes for Graph Operations that provide a neat interface and consistency in practice. 

However, we found that for certain Operations that required multiple inputs (i.e. Merge, Concat) the generated class was incorrect as it took a single input instead of an inputList. 

For example in Merge.java:
```  
public static <T> Merge<T> create(Scope scope, Operand<T> inputs) { // inputs should be a list
    OperationBuilder opBuilder = scope.graph().opBuilder("Merge", scope.makeOpName("Merge"));
    opBuilder.addInput(inputs.asOutput()); //Should be .addInputList() not .addInput()
    return new Merge<T>(opBuilder.build());
}
```

**Describe the expected behavior**
Regarding the code snippet above, we need `inputs` to be a list rather than a single Operand and `.addInput()` should be `.addInputList()`. 

**Code to reproduce the issue**

1. This code below compiles but errors during the run with the error below.
```
    Constant constant1 = Constant.create(scope, 1);
    Enter enter = Enter.create(scope, constant1, "Loop");
    Merge.create(scope, enter);
```
```
[error] Exception in thread "main" java.lang.IllegalArgumentException: Single tensor passed to 'inputs', expected list while building NodeDef 'Merge' using Op<name=Merge; signature=inputs:N*T -> output:T, value_index:int32; attr=T:type; attr=N:int,min=1>
```

2. On the other hand the below code does not compile:
```
    Constant constant1 = Constant.create(scope, 1);
    Enter enter = Enter.create(scope, constant1, "Loop");
    Operand[] inputList = {enter};
    Merge.create(scope, inputList);
```
```
[info] Compiling 1 Java source to /Users/irenedea/linreg/target/scala-2.12/classes ...
[error] /Users/irenedea/linreg/src/main/java/GraphBuilder.java:33:1: method create in class org.tensorflow.op.core.Merge<T> cannot be applied to given types;
[error]   required: org.tensorflow.op.Scope,org.tensorflow.Operand<T>
[error]   found: org.tensorflow.op.Scope,org.tensorflow.Operand[]
[error]   reason: cannot infer type-variable(s) T
[error]     (argument mismatch; org.tensorflow.Operand[] cannot be converted to org.tensorflow.Operand<T>)
[error]     Merge.create(scope, inputList);
```

**Other info / logs**
Created a hack around this issue by building the merge from scratch with the opBuilder, but would be best to have the generated code to be correct as it provides a neat interface :)
```
  public <T> Operation merge(Scope scope, Output<T>[] inputs) {
    return graph.opBuilder("Merge", scope.makeOpName("Merge"))
            .addInputList(inputs).build();
  }
```
```
    Constant constant1 = Constant.create(scope, 1);
    Enter enter = Enter.create(scope, constant1, "Loop");
    Output[] inputList = {enter.asOutput()};
    merge(scope, inputList);
```

**Comment by irenedea** [comment]
We found the issue in tensorflow/java/src/gen/cc/op_specs.cc in the TypeOf function. 

The fix is a super simple change, moving an if statement block from the beginning to the end of the function. See the fix here: 
https://github.com/irenedea/tensorflow/commit/264648c4b30479a4f110bd6ce66830c2963ee4ae
"Move initial iterable check in TypeOf function from the beginning to the end of the function. (The check checks to see if the Op has an attribute N, which would signify the need for a list rather than a single input.) Previously, the iterable bool was overwritten by a later call to TypesOf in the TypeOf function. This caused iterable to be incorrectly false for Ops with attributes T: type and N: int, such as Merge and Concat."

**Comment by irenedea** [comment]
PR here: https://github.com/tensorflow/tensorflow/pull/24813

**Comment by asimshankar** [comment]
Thanks for the report and even more for the fix! Will take a look at the PR.
FYI @karllessard

**Comment by asimshankar** [comment]
For the record, fixed by #24813
Thanks @samdow @irenedea @melissagrueter
